### ISSUE TITLE
springBootUtility command

### ISSUE THREAD

**Issue by lauracowen** [OpenLiberty/docs#776]
https://www.ibm.com/support/knowledgecenter/en/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_command_sbutility.html

When published, add a comment to issue so they can update the guide link: https://github.com/OpenLiberty/guide-spring-boot/issues/21

**Comment by Rwalls1** [comment]
@aguibert Please review: https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/springbootUtility-help.html
https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/springbootUtility-thin.html
What other information is needed for each respective command page? I'm a bit confused about which information from the KC page is relevant to each command: https://www.ibm.com/support/knowledgecenter/en/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_command_sbutility.html

**Comment by aguibert** [comment]
hi @Rwalls1, I'm not familiar with the springbootUtility command, so I'm going to tag @tjwatson for review instead on this one

**Comment by tjwatson** [comment]
The `thin` command should include the various options that can be used with that command:

```
Options:

--sourceAppPath=path
The path of the source application file to thin.
This option is required.

--parentLibCachePath=directory
The directory path of the parent read-only library cache. The parent library cache is searched first to locate existing libraries. If the library is not found, the library is stored in the writable library cache that is specified by --targetLibCachePath. If this option is not specified, no parent library cache is searched.
This option is not required.

--targetLibCachePath=directory
The directory path that is used to save the library cache. If this option is not specified, a lib.index.cache directory is created in the parent directory of the source application.
This option is not required.

--targetThinAppPath=file
The path that is used to save the thin application file. If this option is not specified, a new file is created with the .spring extension in the parent directory of the source application.
This option is not required.
```

**Comment by Rwalls1** [comment]
@tjwatson Thank you, do you know what additional information is needed for the help command?

**Comment by tjwatson** [comment]
So far there is only one additional command `thin`.  You could follow the `help` command for the `server` script:
https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/server-help.html

## springBootUtility help command
Prints help information for the springBootUtility command, with details about its actions and options.

## Usage examples
Display help information for the `springbootUtility` command:

```
springbootUtility help
```
Display help information for the `thin` action:

```
springbootUtility help thin
```

## Syntax
Run the command from the `path_to_liberty/bin` directory.

```
springbootUtility help actionName
```
**_actionName_**
The name of the `springbootUtility` command action that you want information about.

**Comment by Rwalls1** [comment]
@tjwatson Thanks, Please review: 
https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/springbootUtility-thin.html
https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/springbootUtility-help.html

**Comment by tjwatson** [comment]
The springBootUtility `thin` command doesn't make much sense without the information about why spring boot applications should be thinned:  https://www.ibm.com/support/knowledgecenter/en/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/twlp_thin_sb_applications.html

We either need to put this information in with the springbootUtility documentation or it needs to be a separate section under DEPLOYMENT.  My preference would be to have this information in a separate section under DEPLOYMENT.

At a minimum for the `springbootUtility-thin.html` we should add a usage example from KC:

```
Invoke the springBootUtility command with the necessary options to create the thin Spring Boot application in the correct dropins folder of the helloserver server configuration and to cache the dependencies to the usr/shared/resources/lib.index.cache directory.
<wlp>/bin/springBootUtility thin \
    --sourceAppPath=hellospringboot.jar \
    --targetLibCachePath=<wlp>/usr/shared/resources/lib.index.cache \
    --targetThinAppPath=<wlp>/usr/servers/helloserver/dropins/spring/hellospringboot.jar
```

**Comment by Rwalls1** [comment]
@tjwatson Thanks, please review the latest updates, we will discuss having the information in a separate section: 
https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/springbootUtility-thin.html
https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/springbootUtility-help.html

**Comment by tjwatson** [comment]
The latest looks good for the `thin` and `help` springBootUtility docs.  Let me know when I can review the separate section on configuring/deploying Spring Boot applications.

**Comment by Charlotte-Holt** [comment]
@Rwalls1 Content review feedback:

https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/springbootUtility-thin.html
- [x] Period after the first intro para
- [x] Need colons leading in to your usage example
- [x] Regarding the **Usage examples** section - I would pull out the more general information about the springBootUtility command into the top section. The actual examples, I think, should be very applied. See the [server debug command](https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/server-debug.html) for examples. 
- [x] Monospace "springBootUtility" in "Invoke the springBootUtility command..."
- [x] Need to include the options as part of the syntax - see the [server package command](https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/server-package.html)
- [x] In options table, monospace "--targetLibCachePath", "lib.index.cache", ".spring"
- [x] Update "by --targetLibCachePath." to "by the `--targetLibCachePath` option."

https://draft-openlibertyio.mybluemix.net/docs/21.0.0.1/reference/command/springbootUtility-help.html
- [x] Period after the first intro para
- [x] Make springBootUtility monospaced in first para
- [x] "boot" needs to be capitalized in the syntax example

**Comment by Rwalls1** [comment]
@Charlotte-Holt Thanks, I've addressed your feedback

**Comment by ManasiGandhi** [comment]
@rw2513  #peer reveiw The topic looks good to me.

**Comment by Charlotte-Holt** [comment]
Per @chirp1's note in https://github.com/OpenLiberty/docs/issues/2008, the [command overview topic](https://github.com/OpenLiberty/docs/blob/draft/modules/reference/pages/command/command-overview.adoc) needs to be updated to add the `springBootUtility` command when these topics are ready to publish.

**Comment by Rwalls1** [comment]
Staging:
https://staging-openlibertyio.mybluemix.net/docs/latest/reference/command/springbootUtility-help.html
https://staging-openlibertyio.mybluemix.net/docs/latest/reference/command/springbootUtility-thin.html

**Comment by Rwalls1** [comment]
Doc has been merged to Vnext and will publish with 21.0.0.6

**Comment by chirp1** [comment]
@Rwalls1 Hi Richard, the docs look nice! I have a few comments:
- The SpringBootUtility help command topic looks good.

- I see that the commands overview topic at https://staging-openlibertyio.mybluemix.net/docs/21.0.0.7/reference/command/command-overview.html#springBootUtility has links to your SpringBootUtility thin command topic. Good!

Comments for the SpringBootUtility thin command topic:
- Use a complete sentence for the intro sentence. Some of the other commands in OL use complete sentences for the intro sentence.
- Add to your topic right after the intro the paragraph in the "About" section at https://www.ibm.com/docs/en/was-liberty/base?topic=applications-springbootutility-command (Rework the paragraph as necessary.)
- I'd say, leave off "options One or more optional parameters." The WL topics don't use the phrase. Calling options both options and parameters muddies what they are. I'm gathering that some of the options, such as --sourceAppPath=path, are required. So, saying that something is optional when it is required doesn't make sense.
- Use terms consistently. "path_to_liberty" is consistently used in the command topics. For the example, replace "wlp" with "path_to_liberty".

**Comment by Rwalls1** [comment]
@chirp1 Thanks, I've addressed your feedback: https://docs-draft-openlibertyio.mybluemix.net/docs/latest/reference/command/springbootUtility-thin.html
**Add to your topic right after the intro the paragraph in the "About" section at https://www.ibm.com/docs/en/was-liberty/base?topic=applications-springbootutility-command (Rework the paragraph as necessary.)**
We discussed in slack that this suggestion doesn't quite align with the current strategy for the command topics and that the paragraph that you mentioned is already covered in the related guide that is referenced on the command overview page.
**Use terms consistently. "path_to_liberty" is consistently used in the command topics. For the example, replace "wlp" with "path_to_liberty"**
After discussing with the other OL writers and reviewing examples from other topics, I discovered that the actual path should be "path_to_liberty/wlp/bin". This change has also been made in the other command reference topics to maintain consistency.

**Comment by chirp1** [comment]
@Rwalls1 Hi Richard, 
- We discussed with David yesterday that adding in the type of info in "About" would generally be appropriate in any of the command docs. So, go ahead and add it in for this command topic. Rework the addition as appropriate.
- Since you, David, and I decided that we want to italicize variables, go ahead and do it in your topic. The updates for the rest of the topics can be dealt with in some other issue specific to changing variables to italics.
- I spotted one other item. Change "...with the springBootUtility command a..." to "...with the springBootUtility thin command a...". (Add "thin" to be consistent with what you put in the intro sentence for the topic.)

The rest of your updates  look good! Thanks!

**Comment by Rwalls1** [comment]
@chirp1 Thanks, I've made those further updates: https://draft-openlibertyio.mybluemix.net/docs/21.0.0.7/reference/command/springbootUtility-thin.html

**Comment by chirp1** [comment]
@Rwalls1 Hi Richard, 

- For the code example, have you tried italicizing "path_to_liberty" for "targetLibCachePath" and "targetThinAppPath"? If you haven't would you give it a try? It might or might not work- possibly too complex for AsciiDoc to do italics with "=" on the left and "/" on the right. Let me know how it goes.
- Would you add a "See also" section that has a link to the Spring Boot guide?

The rest of the updates look good. Thanks!

**Comment by Rwalls1** [comment]
@chirp1 Thanks, I've made those updates: https://draft-openlibertyio.mybluemix.net/docs/21.0.0.7/reference/command/springbootUtility-thin.html

**Comment by chirp1** [comment]
@Rwalls1 Hi Richard, 

- The italics update looks good.
- The "See also" link to the guide is broken for me, but I can get to it on the draft server at https://draft-openlibertyio.mybluemix.net/guides/spring-boot.html . Is the link something to be fixed, or will the link resolve on its own when the topic is published? Let me know. Thanks!

**Comment by Rwalls1** [comment]
@chirp1 Thanks, the link should be working now: https://draft-openlibertyio.mybluemix.net/docs/latest/reference/command/springbootUtility-thin.html

**Comment by chirp1** [comment]
@Rwalls1 The topic looks great! Moving to Ready to Publish.

**Comment by Rwalls1** [comment]
Staging: https://staging-openlibertyio.mybluemix.net/docs/21.0.0.7/reference/command/springbootUtility-thin.html

**Comment by Rwalls1** [comment]
The latest updates have been merged to vNext and will publish with 21.0.0.7.
